Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonGun.cs

```diff
@@ -275,14 +294,38 @@ private void HarpoonFullyReloaded()
     {
         _harpoonFiringState = EHarpoonFiringState.Ready;
         _harpoonOnGun.SetActive(true);
+        RuntimeSfxManager.APlayOneShotSfx?
+            .Invoke(FmodSfxEvents.Instance.HarpoonReload, gameObject.transform.position);
 
-        if(_isFocusButtonHeld)
+        if (_isFocusButtonHeld)
         {
             StartFocusingHarpoon();
         }
 
         PlayerManager.Instance.InvokeOnHarpoonReloadedEvent();
     }
+
+    /// <summary>
+    /// Restocks harpoons to full from the given ammo rack
+    /// </summary>
+    /// <param name="ammoRack"> The ammo rack to take harpoons from </param>
+    private void RestockHarpoons(AmmoRackInteractable ammoRack)
+    {
+        int missingAmmo = _maxAmmo - _currentReserveAmmo;
+        if (_harpoonFiringState == EHarpoonFiringState.Ready)
+        {
+            missingAmmo--;
+        }
+
+        int numHarpoons = ammoRack.GetNumHarpoons();
+
+        int targetAmmo = numHarpoons < missingAmmo ? numHarpoons : missingAmmo;
```

Comment:
woah I have never seen something like this before. what does it do? Can you leave a comment above it for any other engineer or designer who sees this function.